Firestore/core/src/remote/firebase_platform_logging_apple.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_REMOTE_FIREBASE_PLATFORM_LOGGING_APPLE_H_
+#define FIRESTORE_CORE_SRC_REMOTE_FIREBASE_PLATFORM_LOGGING_APPLE_H_
+
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++."
+#endif  // !defined(__OBJC__)
+
+#include <string>
+
+#include "Firestore/core/src/remote/firebase_platform_logging.h"
+
+@class FIRApp;
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+class FirebasePlatformLoggingApple : public FirebasePlatformLogging {
+ public:
+  explicit FirebasePlatformLoggingApple(FIRApp* app);
```

Comment:
Objective-C types should have their nullability defined, either by wrapping in `NS_ASSUME_NONNNULL_BEGIN`/`END` or by annotating the pointers directly (`FIRApp* _Nonnull app`) .